### ISSUE TITLE
How pass extra parameters to Axios

### ISSUE THREAD

**Issue by OliverDudgeon** [anymaniax/orval#115]
I want to track progress of my POST request. With Axios I  would usually pass an `onUploadProgress` parameter to `axios.post(...)`. With orval I would want to do this from the function/hook it generates. Is this currently possible? This seems to be a more general issue with other parameters of `AxiosRequestConfig` not being exposed to the user of the orval. 

## What versions are you using?
**Package Version: 5.0.0-alpha.9
**Browser Version: Firefox 87

**Comment by anymaniax** [comment]
Hello, for the moment the only way to do that is to use a [mutator](https://orval.dev/reference/configuration#mutator) you can add your mutator globally or on a specify [operation](https://orval.dev/reference/configuration#operations)

**Comment by anymaniax** [comment]
I will think about a way to add this feature. Would it be ok for you to add the option in the Orval config or you need to be able to do it dynamically at runtime? With the mutator I am http client agnostic and I want to keep it like that.

**Comment by OliverDudgeon** [comment]
I think my use case will be limited to the options exposed in the definition of the client: so just those extra options defined in `AxiosRequestConfig`. Nothing dynamically created.

**Comment by anymaniax** [comment]
You can do a global mutator like [here](https://github.com/anymaniax/orval/blob/master/samples/react-app-with-react-query/src/api/mutator/custom-instance.ts) and check if the method is `POST` add the `onUploadProgress` parameter to the axios instance.

**Comment by anymaniax** [comment]
Is it ok for you if with the last argument you have the possibility to override the `AxiosRequestConfig`? But if you don't want to pass a `QueryConfig` you will need to pass undefined for that one to have the possibility to override the `AxiosRequestConfig`

**Comment by OliverDudgeon** [comment]
I think that would work. But, would passing a single object with queryConfig and axiosRequestConfig keys not be suitable (other than it being a breaking change)?

**Comment by anymaniax** [comment]
yes, maybe it's a better solution I will have time to do that this weekend normally

**Comment by anymaniax** [comment]
Hello @OliverDudgeon, I published a new alpha with the possibility to pass an `AxiosRequestConfig` if you don't use a mutator. I have an idea in the case you use mutator but it will need me more time to add it.

**Comment by anymaniax** [comment]
I released a new version of orval which also handle a second argument with a mutator
